source/mainnet/smart-contracts/general/introduction.rst

```diff
@@ -88,6 +88,22 @@ module <contract-module>`. After this a smart contract can be *initialized* to
 obtain a :ref:`smart contract instance <contract-instances>`. Finally a smart
 contract instance can be repeatedly updated according to its own logic.
 
+.. image:: images/smart-contract-lifecycle.png
+    :width: 100%
+    :alt: flow diagram with different actions
```

Comment:
Nitpick: the names of all boxes on the diagram start with a capital letter, apart from "edit" :)

Reply:
This is intentional since there is no edit command.